Source/DataSources/KmlDataSource.js

```diff
@@ -1474,27 +1628,370 @@ define([
+                if (argFirst === '?' || argFirst === '&') {
+                    arg.splice(0, 1);
+                }
+
+                var last = result.slice(-1);
+                if (result.length > 0 && last !== '?' && last !== '&') {
+                    result += '&';
+                }
+                result += arguments[i];
+            }
+        }
+
+        return result;
+    }
+
+    function appendQueryString(url, queryString) {
+        if (defined(queryString) && queryString.length > 0) {
+            var c = (url.indexOf('?') === -1) ? '?' : '&';
+            return url + c + queryString;
+        }
+
+        return url;
+    }
+
+    function processNetworkLinkQueryString(camera, canvas, queryString, viewBoundScale, bbox) {
+        function fixLatitude(value) {
+            if (value < -CesiumMath.PI_OVER_TWO) {
+                return -CesiumMath.PI_OVER_TWO;
+            } else if (value > CesiumMath.PI_OVER_TWO) {
+                return CesiumMath.PI_OVER_TWO;
+            }
+            return value;
+        }
+
+        function fixLongitude(value) {
+            if (value > CesiumMath.PI) {
+                return value - CesiumMath.TWO_PI;
+            } else if (value < -CesiumMath.PI) {
+                return value + CesiumMath.TWO_PI;
+            }
+
+            return value;
+        }
+
+        if (defined(camera)) {
+            var wgs84 = Ellipsoid.WGS84;
+            var centerCartesian;
+            var centerCartographic;
+
+            bbox = defaultValue(bbox, computeViewRectangle(camera, canvas));
+            if (defined(canvas)) {
+                scratchCartesian2_1.x = canvas.clientWidth * 0.5;
+                scratchCartesian2_1.y = canvas.clientHeight * 0.5;
+                centerCartesian = camera.pickEllipsoid(scratchCartesian2_1, wgs84, scratchCartesian3_1);
+            }
+
+            if (defined(centerCartesian)) {
+                centerCartographic = wgs84.cartesianToCartographic(centerCartesian, scratchCartographic);
+            } else {
+                centerCartographic = Rectangle.center(bbox, scratchCartographic);
+                centerCartesian = wgs84.cartographicToCartesian(centerCartographic);
+            }
+
+
+            if (defined(viewBoundScale) && !CesiumMath.equalsEpsilon(viewBoundScale, 1.0, CesiumMath.EPSILON9)) {
+                var newHalfWidth = bbox.width * viewBoundScale * 0.5;
+                var newHalfHeight = bbox.height * viewBoundScale * 0.5;
+                bbox = new Rectangle(fixLongitude(centerCartographic.longitude - newHalfWidth),
+                    fixLatitude(centerCartographic.latitude - newHalfHeight),
+                    fixLongitude(centerCartographic.longitude + newHalfWidth),
+                    fixLatitude(centerCartographic.latitude + newHalfHeight)
+                );
+            }
+
+            queryString = queryString.replace('[bboxWest]', CesiumMath.toDegrees(bbox.west).toString());
```

Comment:
How often is this code called?  Would it be worth having a big regex for the replacement instead of all of these manual replacements?